integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala

```diff
@@ -96,6 +96,25 @@ class UpdateCarbonTableTestCase extends QueryTest with BeforeAndAfterAll {
     sql("""drop table if exists iud.dest22""")
   }
 
+  test("update with subquery with more than one value for key") {
+    sql("drop table if exists t1")
+    sql("drop table if exists t2")
+    sql("create table t1 (age int, name string) stored by 'carbondata'")
+    sql("insert into t1 select 1, 'aa'")
+    sql("insert into t1 select 2, 'aa'")
+    sql("create table t2 (age int, name string) stored by 'carbondata'")
+    sql("insert into t2 select 1, 'Andy'")
+    sql("insert into t2 select 2, 'Andy'")
+    val exception = intercept[RuntimeException] {
+      sql("update t1 set (age) = (select t2.age from t2 where t2.name = 'Andy') where t1.age = 1 ").show(false)
```

Comment:
please add more cases, for example:
`update t1 set () = (select .. from t2 where t2.id = t1.id) where ...`

Reply:
ok, I will add test later and if required we can add carbon property 
a. by default this property is true (this may have performance degrade) and carbon validate each update. 
user can set this property to false if he don't want validation and he knows what he is doing.

what do you think ?

Reply:
OK.

Reply:
added carbon property, updated document and test case